tests/main/buildmode/task.yaml

```diff
@@ -0,0 +1,22 @@
+summary: Test that we use the right buildmode per arch
+
+systems: [ubuntu-1*, ubuntu-2*]
+
+execute: |
+    # `objdump -f` will report "DYNAMIC" for binaries build with
+    # -buildmode=pie
+    needle="DYNAMIC"
+    
+    # 32bit arches (like armhf/i386) are build without -buildmode=pie
+    # as it breaks too many things if we do.
+    if [ "$(getconf LONG_BIT)" = 32 ]; then
+        # and "EXEC_P" when the buildmode is "normal"
+        needle="EXEC_P"
+    fi
+
+    # check /usr/bin/snap
+    objdump -f /usr/bin/snap | MATCH "$needle"
+    # and the helpers in /usr/lib/snapd
+    for p in snapd snap-repair; do
```

Comment:
should we check everything that is a binary under /usr/lib/snapd? I think it would be worthwile to at least include snap-confine since it's privileged and written in C.

Reply:
Sounds good, lets do that in a followup!